block/manager.go

```diff
@@ -566,17 +560,17 @@ func (m *Manager) getCommit(header types.Header) (*types.Commit, error) {
 func (m *Manager) IsProposer() (bool, error) {
 	m.lastStateMtx.RLock()
 	defer m.lastStateMtx.RUnlock()
-	// if proposer is not set, assume self proposer
-	if m.lastState.Validators.Proposer == nil {
-		return true, nil
-	}
 
 	signerPubBytes, err := m.proposerKey.GetPublic().Raw()
 	if err != nil {
 		return false, err
 	}
 
-	return bytes.Equal(m.lastState.Validators.Proposer.PubKey.Bytes(), signerPubBytes), nil
+	if len(m.genesis.Validators) == 1 {
```

Comment:
i recommend moving this check to `NewManager` as you don't need to check this on every block proposal. Just check the negation of this in the constructor then you can assume 1 validator here.